tests/utils/sampler_test.py

```diff
@@ -1,36 +1,35 @@
-from pathlib import Path
-
 import numpy as np
 import pytest
+import torch
 from scipy.stats import chisquare
 
-from fl4health.utils.dataset import MnistDataset
+from fl4health.utils.dataset import SyntheticDataset
 from fl4health.utils.random import set_all_random_seeds, unset_all_random_seeds
 from fl4health.utils.sampler import DirichletLabelBasedSampler, MinorityLabelBasedSampler
 
 
+def construct_synthetic_dataset() -> SyntheticDataset:
+    # set seed for creation
+    torch.manual_seed(42)
+    random_inputs = torch.rand((10000, 3, 3))
+    # Note high is exclusive here
+    random_labels = torch.randint(low=0, high=10, size=(1000, 1))
+    # unset seed thereafter
+    torch.seed()
+    return SyntheticDataset(random_inputs, random_labels)
+
+
+SYNTHETIC_DATASET = construct_synthetic_dataset()
```

Comment:
This dataset will be reused for **all** tests herein

Reply:
I am confused why random_inputs and random_labels size are not matching?

Reply:
lol...that's a good catch! I lost a zero somehow on the labels. I added an assertion in the synthetic dataset that will fail if this happens again. Apparently there is no assertion in the pytorch dataset class that forces this. I'll push the fix in a minute after the tests run.